crates/re_build_info/src/crate_version.rs

```diff
@@ -298,11 +455,85 @@ fn test_compatibility() {
     assert!(are_compatible("0.2.0", "0.2.0"));
     assert!(are_compatible("0.2.0", "0.2.1"));
     assert!(are_compatible("1.2.0", "1.3.0"));
-    assert!(!are_compatible("0.2.0", "1.2.0"));
-    assert!(!are_compatible("0.2.0", "0.3.0"));
+    assert!(
+        !are_compatible("0.2.0", "1.2.0"),
+        "Different major versions are incompatible"
+    );
+    assert!(
+        !are_compatible("0.2.0", "0.3.0"),
+        "Different minor versions are incompatible"
+    );
     assert!(are_compatible("0.2.0-alpha.0", "0.2.0-alpha.0"));
+    assert!(are_compatible("0.2.0-rc.0", "0.2.0-rc.0"));
+    assert!(
+        !are_compatible("0.2.0-rc.0", "0.2.0-alpha.0"),
+        "Rc and Alpha are incompatible"
+    );
+    assert!(
+        !are_compatible("0.2.0-rc.0", "0.2.0-alpha.0+dev"),
+        "Rc and Dev are incompatible"
+    );
+    assert!(
+        !are_compatible("0.2.0-alpha.0", "0.2.0-alpha.0+dev"),
+        "Alpha and Dev are incompatible"
+    );
     assert!(
         !are_compatible("0.2.0-alpha.0", "0.2.0-alpha.1"),
-        "Alphas are always incompatible"
+        "Different alpha builds are always incompatible"
+    );
+    assert!(
+        !are_compatible("0.2.0-rc.0", "0.2.0-rc.1"),
+        "Different rc builds are always incompatible"
+    );
```

Comment:
as per above I wouldn't have expected that, but I suppose it's debatable 🤔. In my mind an RC is "almost a release" means that if doesn't fullfil the compatibility requirements of the release itself we have a bug that needs to be fixed before the final release.


also (less important) missing in the same vain:

```
    assert!(
        are_compatible("0.2.1-rc.0", "0.2.0"),
        "rc builds for minor version changes are compatible"
    );
```

Reply:
So just to clarify, you mean that `0.2.1-rc.0 == 0.2.1-rc.1 == 0.2.1`?

Reply:
I think so, but maybe we should check with the others as well :)